src/modules/hip/kernel/normalize.hpp

```diff
@@ -0,0 +1,1925 @@
+                                             uint *srcMaxDims,
+                                             uint *srcStrides,
+                                             float *meanTensor,
+                                             float *stdDevTensor,
+                                             uint *roiTensor,
+                                             uint *paramShapeTensor,
+                                             uint *paramStridesTensor,
+                                             uint maxParamVolume,
+                                             uint numDims,
+                                             uint maxBufferLength)
+{
+    uint id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    uint id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    uint *roi = &roiTensor[id_z * numDims * 2 + numDims];
+    uint *paramShape = &paramShapeTensor[id_z * numDims];
+    uint *paramStrides = &paramStridesTensor[id_z * numDims];
+    uint srcIdx = id_z * maxBufferLength + id_x;
+    uint paramBase = id_z * maxParamVolume;
+    uint paramIndex = 0;
+
+    if(maxParamVolume > MAX_SHARED_MEMORY_SIZE)
+    {
+        if(id_x >= maxBufferLength)
+            return;
+
+        // validate if id_x is within the roi of input and compute paramIndex if valid
+        for (int i = 0; i < numDims; i++)
+        {
+            uint coord = id_x / srcStrides[i] % srcMaxDims[i];
+            if(coord >= roi[i])
+                return;
+            paramIndex += (maxParamVolume > 1) ? (rpp_hip_mod(coord, paramShape[i]) * paramStrides[i]) : 0;
+        }
+        float val = srcPtr[srcIdx] - meanTensor[paramBase + paramIndex];
+        atomicAdd(&stdDevTensor[paramBase + paramIndex], (val * val));
+    }
+    else
+    {
+
+        if(id_x >= (hipBlockDim_x * hipGridDim_x))
+            return;
+
+        // if number of means needed to compute is within in the max shared memory size
+        // use shared memory for atomic addition to reduce global memory traffic
+        bool isValid = true;
+        for (int i = 0; i < numDims; i++)
+        {
+            uint coord = id_x / srcStrides[i] % srcMaxDims[i];
+            if(coord >= roi[i])
+            {
+                isValid = false;
+                break;
+            }
+            paramIndex += (maxParamVolume > 1) ? (rpp_hip_mod(coord, paramShape[i]) * paramStrides[i]) : 0;
+        }
+
+        extern __shared__ float sh_mem[];
+        sh_mem[hipThreadIdx_x] = 0.0f;
+        __syncthreads();
+
+        if(isValid && id_x < maxBufferLength)
+        {
+            float val = srcPtr[srcIdx] - meanTensor[paramBase + paramIndex];
+            atomicAdd(&sh_mem[paramIndex], (val * val));
+        }
+        __syncthreads();
+
+        if (hipThreadIdx_x < maxParamVolume)
+            atomicAdd(&stdDevTensor[paramBase + hipThreadIdx_x], sh_mem[hipThreadIdx_x]);
+    }
+}
+
+void set_kernel_launch_config_2d(RpptGenericDescPtr srcGenericDescPtr,
```

Comment:
Use a clear sectional delineation to show which ones are device helpers, host helpers, device kernels and kernel executors or other categories.
You can use the formatting from something like box filter that has many divisions - https://github.com/r-abishek/rpp/blob/master/src/modules/hip/kernel/box_filter.hpp#L4